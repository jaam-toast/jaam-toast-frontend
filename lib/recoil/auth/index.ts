import loginState from "./atom";
import isLoggedInState from "./isLoggedInSelector";

export { isLoggedInState };

export default loginState;
