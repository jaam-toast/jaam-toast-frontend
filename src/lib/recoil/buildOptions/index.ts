import { buildOptionsState, buildStepState } from "./atoms";

export { buildOptionsState, buildStepState };
