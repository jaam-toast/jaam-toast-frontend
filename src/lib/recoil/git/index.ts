import { cloneUrlState, gitNamespaceState, gitRepoState } from "./atom";
import { cloneRepoName, gitNamespaceList } from "./selector";

export {
  cloneUrlState,
  gitNamespaceState,
  gitRepoState,
  cloneRepoName,
  gitNamespaceList,
};
