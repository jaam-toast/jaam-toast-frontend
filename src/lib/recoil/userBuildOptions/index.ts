import buildOptionState from "./atoms";

export default buildOptionState;
