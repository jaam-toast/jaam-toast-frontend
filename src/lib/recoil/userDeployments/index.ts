import userDeploymentsState from "./atom";

export default userDeploymentsState;
