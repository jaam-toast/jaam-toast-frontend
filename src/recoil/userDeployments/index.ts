import { userDeploymentsState, selectedProject } from "./atom";

export { userDeploymentsState, selectedProject };
