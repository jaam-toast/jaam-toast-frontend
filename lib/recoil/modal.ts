import { atom } from "recoil";

export const MODAL_TYPES = {
  ModalCreate: "ModalCreate",
  ModalBuild: "ModalBuild",
  ModalDeploy: "ModalDeploy",
  ModalDeleteConfirm: "ModalDeleteConfirm",
  ModalDeleteAlert: "ModalDeleteAlert",
  ModalPreview: "ModalPreview",
  ModalRepoDetails: "ModalRepoDetails",
} as const;

export interface ModalCreateType {
  modalType: typeof MODAL_TYPES.ModalCreate;
  modalProps?: any;
}

export interface ModalBuildType {
  modalType: typeof MODAL_TYPES.ModalBuild;
  modalProps?: any;
}

export interface ModalDeployType {
  modalType: typeof MODAL_TYPES.ModalDeploy;
  modalProps?: any;
}

export interface ModalDeleteType {
  modalType: typeof MODAL_TYPES.ModalDeleteConfirm;
  modalProps?: any;
}
export interface ModalDeleteAlertType {
  modalType: typeof MODAL_TYPES.ModalDeleteAlert;
  modalProps?: any;
}

export interface ModalPreviewType {
  modalType: typeof MODAL_TYPES.ModalPreview;
  modalProps?: any;
}

export interface ModalRepoDetailsType {
  modalType: typeof MODAL_TYPES.ModalRepoDetails;
  modalProps?: any;
}

export type ModalType =
  | ModalCreateType
  | ModalBuildType
  | ModalDeployType
  | ModalDeleteType
  | ModalDeleteAlertType
  | ModalPreviewType
  | ModalRepoDetailsType;

export const modalState = atom<ModalType | null>({
  key: "modalState",
  default: null,
});
