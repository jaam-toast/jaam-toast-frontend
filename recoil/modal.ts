import { atom } from "recoil";

export const MODAL_TYPES = {
  ModalCreate: "ModalCreate",
  ModalBuild: "ModalBuild",
} as const;

export interface ModalCreateType {
  modalType: typeof MODAL_TYPES.ModalCreate;
}

export interface ModalBuildType {
  modalType: typeof MODAL_TYPES.ModalBuild;
}

export type ModalType = ModalCreateType | ModalBuildType;

export const modalState = atom<ModalType | null>({
  key: "modalState",
  default: null,
});
