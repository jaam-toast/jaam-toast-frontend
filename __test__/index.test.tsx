import { render, screen } from "@testing-library/react";

import { RecoilRoot } from "recoil";

import PageLanding from "src/pages/index";

describe("PageLanding", () => {
  it("renders PageLanding unchanged", () => {
    const { container } = render(
      <RecoilRoot>
        <PageLanding />
      </RecoilRoot>,
    );

    expect(container).toMatchSnapshot();
  });

  it("PageLanding contains welcomText and loginButton", async () => {
    const component = (
      <RecoilRoot>
        <PageLanding />
      </RecoilRoot>
    );

    render(component);

    const welcomeText = await screen.findByText(/deploy your own project/i);
    const loginButton = await screen.findByText(/with github/i);
    expect(welcomeText).toBeTruthy();
    expect(loginButton).toBeTruthy();
  });
});
