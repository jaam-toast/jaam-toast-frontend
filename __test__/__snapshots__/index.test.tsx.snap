// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLanding renders PageLanding unchanged 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-disableGutters css-idqxm-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1wfmdn2"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionRelative css-th72da-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-10klw3m"
          >
            <button
              aria-label="brand-logo"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Jaamtoast logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiTypography-root MuiTypography-h4 css-vlov06-MuiTypography-root"
          >
            /
          </div>
          <div
            class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
          >
            Jaam-Toast
          </div>
        </div>
      </header>
    </div>
    <div
      class="MuiBox-root css-1h1r4d"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1bojnzi-MuiTypography-root"
      >
        Deploy your own project.
      </h2>
      <div
        class="MuiBox-root css-1mtqcvq"
      >
        <div
          class="MuiBox-root css-1h5m10"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
          >
            No projects, yet!
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-130dmu7-MuiTypography-root"
          >
            Create a project by importing a Git repository.
          </p>
        </div>
        <div
          class="MuiBox-root css-1xhj18k"
        >
          <div
            class="MuiBox-root css-jhhg0i"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="Develop"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                  height="130"
                  src="develop_code.jpg"
                />
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <div
                    class="MuiBox-root css-9o5tfo"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-elhp89-MuiSvgIcon-root"
                      data-testid="ChangeHistoryIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7.77 18.39 18H5.61L12 7.77M12 4 2 20h20L12 4z"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-136m5w4-MuiTypography-root"
                    >
                      Develop
                    </p>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-jhhg0i"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="Programming"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                  height="130"
                  src="programming_react.jpg"
                />
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <div
                    class="MuiBox-root css-9o5tfo"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-elhp89-MuiSvgIcon-root"
                      data-testid="ChangeHistoryIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7.77 18.39 18H5.61L12 7.77M12 4 2 20h20L12 4z"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-136m5w4-MuiTypography-root"
                    >
                      Programming
                    </p>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-jhhg0i"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="Co-working"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                  height="130"
                  src="coworking.jpg"
                />
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <div
                    class="MuiBox-root css-9o5tfo"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-elhp89-MuiSvgIcon-root"
                      data-testid="ChangeHistoryIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7.77 18.39 18H5.61L12 7.77M12 4 2 20h20L12 4z"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-136m5w4-MuiTypography-root"
                    >
                      Co-working
                    </p>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-jhhg0i"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-pgkxs3-MuiPaper-root-MuiCard-root"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
                tabindex="0"
                type="button"
              >
                <img
                  alt="Deploy"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                  height="130"
                  src="github.jpg"
                />
                <div
                  class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
                >
                  <div
                    class="MuiBox-root css-9o5tfo"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-elhp89-MuiSvgIcon-root"
                      data-testid="ChangeHistoryIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7.77 18.39 18H5.61L12 7.77M12 4 2 20h20L12 4z"
                      />
                    </svg>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-136m5w4-MuiTypography-root"
                    >
                      Deploy
                    </p>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-1qd84cd-MuiTypography-root"
      >
        Every deployment from the 2022 edition of 
        <b>
          jaam-toast.
        </b>
      </h6>
      <div
        class="MuiBox-root css-hpgf8j"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-qkmkh7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ddhjnx-MuiSvgIcon-root"
            data-testid="GitHubIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
            />
          </svg>
          Log in with GitHub
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
